contracts/contracts/extensions/payout/SimplePayout.sol

```diff
@@ -0,0 +1,190 @@
+/// and a MACI Tally contract
+contract SimplePayout is Ownable(msg.sender) {
+  /// @notice use safe ERC20 functions
+  using SafeERC20 for IERC20;
+
+  /// @notice WETH address
+  IWETH public immutable WETH;
+
+  /// @notice the project registry
+  ISimpleProjectRegistry public immutable projectRegistry;
+
+  /// @notice the payout token
+  address public immutable payoutToken;
+
+  /// @notice the tally contract
+  ITally public immutable tally;
+
+  /// @notice keep track of which users have already been paid
+  mapping(uint256 => bool) public paidProjects;
+
+  /// @notice the total amount of tokens available to be paid out
+  uint256 public immutable totalAmount;
+
+  /// @notice the total amount of voice credits spent
+  uint256 public immutable totalVoiceCreditsSpent;
+
+  /// @notice keep track of how many users have been paid
+  uint256 public paid;
+
+  /// @notice Custom errors
+  error InvalidSpentVoiceCredits();
+  error ProjectAlreadyPaid();
+  error InvalidProof();
+  error NotAllProjectsPaid();
+  error NotEnoughEther();
+  error TransferFailed();
+
+  /// @notice Create a new instance of the payout contract
+  /// @param _projectRegistry the address of the project registry
+  /// @param _token the address of the payout token
+  /// @param _tally the address of the tally contract
+  /// @param _totalAmount the total amount of tokens available to be paid out
+  /// @param _totalSpent the total amount of voice credits spent
+  /// @param _totalSpentSalt the salt of the spent amount
+  /// @param _resultCommitment the commitment of the results
+  /// @param _perVOSpentVoiceCreditsHash the hash of the spent voice credits
+  constructor(
+    address _projectRegistry,
+    address _token,
+    address _tally,
+    uint256 _totalAmount,
+    uint256 _totalSpent,
+    uint256 _totalSpentSalt,
+    uint256 _resultCommitment,
+    uint256 _perVOSpentVoiceCreditsHash,
+    address _weth
+  ) payable {
+    projectRegistry = ISimpleProjectRegistry(_projectRegistry);
+    payoutToken = _token;
+    tally = ITally(_tally);
+    WETH = IWETH(_weth);
+
+    // set the total amount of tokens available to be paid out
+    totalAmount = _totalAmount;
+
+    // ensure that the total amount of voice credit spent is correct
+    if (
+      !ITally(_tally).verifySpentVoiceCredits(
+        _totalSpent,
+        _totalSpentSalt,
+        _resultCommitment,
+        _perVOSpentVoiceCreditsHash
+      )
+    ) revert InvalidSpentVoiceCredits();
+
+    // set the total amount of voice credits spent
+    totalVoiceCreditsSpent = _totalSpent;
+  }
+
+  /// @notice Deposit the amount of tokens to the contract
+  /// @dev This function is only callable by the owner
+  function deposit() external payable onlyOwner {
+    if (payoutToken != address(0)) {
+      // transfer the ERC20 token amount
+      IERC20(payoutToken).safeTransferFrom(msg.sender, address(this), totalAmount);
+    }
```

Comment:
If contract is deployed with `payoutToken = address(0)` this deposit will take no effect.
Is it expected behaviour?

Reply:
the function is payable so if payoutToken = address(0) then it just accepts the ether. 

but actually we should have a check that msg.value == totalAmount so I'll add that